extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -415,13 +414,14 @@ export class AmpAnalytics extends AMP.BaseElement {
     }
     const key = this.expandTemplate_(spec['sampleOn'], trigger);
 
-    return urlReplacementsFor(this.getWin()).expand(key).then(key => {
-      const digest = this.sha384_(key);
-      if (digest[0] % 100 < spec['threshold']) {
-        return resolve;
-      }
-      return Promise.resolve(false);
-    });
+    const keyPromise = urlReplacementsFor(this.getWin()).expand(key);
+    const cryptoPromise = cryptoFor(this.getWin());
+    return Promise.all([keyPromise, cryptoPromise])
+        .then(([key, crypto]) => {
```

Comment:
the output this will produce is pretty bad btw, closure compiler will start using Symbol's and iterator protocol. (even babel will)